src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -40,8 +44,9 @@ public class AdminController : Controller
             ILogger<AdminController> logger,
             IStringLocalizer<AdminController> stringLocalizer,
             IUserAssetFolderNameProvider userAssetFolderNameProvider,
-            IChunkFileUploadService chunkFileUploadService
-            )
+            IChunkFileUploadService chunkFileUploadService,
+            IFileVersionProvider fileVersionProvider,
+            IServiceProvider serviceProvider)
```

Comment:
This may not be needed

Reply:
For this and your edits @hishamco: an `IMediaFileStoreCache` instance will only be registered when using a remote media storage, i.e. Azure or AWS. Otherwise, trying to inject it will cause an "Unable to resolve service for type 'OrchardCore.Media.IMediaFileStoreCache'" exception. This is why we need to inject `IServiceProvider` here and resolve it in an optional way.

Reply:
Got it. It sounds like a bad design in how we register `IMediaFileStoreCache`. Any way, this seems to be how we are dealing with this service

Reply:
That can be useful but I wouldn't refactor this for this PR (all Media modules depend on the current behavior).

Reply:
I might do it later